pkg/controller/proxyconfig/controller.go

```diff
@@ -48,6 +51,21 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// We only care about a configmap source with a specific name/namespace,
+	// so filter events before they are provided to the controller event handlers.
+	pred := predicate.Funcs{
+		UpdateFunc:  func(e event.UpdateEvent) bool { return handleConfigMap(e.MetaNew) },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return handleConfigMap(e.Meta) },
+		CreateFunc:  func(e event.CreateEvent) bool { return handleConfigMap(e.Meta) },
+		GenericFunc: func(e event.GenericEvent) bool { return handleConfigMap(e.Meta) },
+	}
+
+	// Watch for changes to the additional trust bundle configmap.
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, pred)
```

Comment:
operator-fwk doesn't let you setup the watch on a specific namespace?

Reply:
@bparees CNO uses the controller-runtime pkg. Cluster Ingress Operator also uses this pkg and follows the same [pattern](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/certificate-publisher/controller.go#L79-L86) of using [predicates](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/event-filtering.md) to filter events from being transformed into requests that get reconciled. The predicate [implemented](https://github.com/openshift/cluster-network-operator/pull/271/files#diff-6cc3b6cf53aa0e5d6919e9943b4b174aR80) in this PR will filter the `ConfigMap` source to namespace `openshift-config`.

Reply:
yeah, just thought we could be more efficient by scoping to the namespace.  but this is fine.